lib/ntlmauth/ntlmauth.h

```diff
@@ -32,24 +32,25 @@ extern "C" {
 #define NTLM_REQUEST_NON_NT_SESSION_KEY     0x400000
 
 /* NTLM error codes */
-#define NTLM_ERR_INTERNAL         -3
-#define NTLM_ERR_BLOB             -2
-#define NTLM_ERR_BAD_PROTOCOL     -1
-#define NTLM_ERR_NONE              0    /* aka. SMBLM_ERR_NONE */
-/* codes used by smb_lm helper */
-#define NTLM_ERR_SERVER            1    /* aka. SMBLM_ERR_SERVER   */
-#define NTLM_ERR_PROTOCOL          2    /* aka. SMBLM_ERR_PROTOCOL */
-#define NTLM_ERR_LOGON             3    /* aka. SMBLM_ERR_LOGON    */
-#define NTLM_ERR_UNTRUSTED_DOMAIN  4
-#define NTLM_ERR_NOT_CONNECTED     10
-/* codes used by mswin_ntlmsspi helper */
-#define NTLM_SSPI_ERROR         1
-#define NTLM_BAD_NTGROUP        2
-#define NTLM_BAD_REQUEST        3
-/* TODO: reduce the above codes down to one set non-overlapping. */
+enum class NtlmError : int
+{
+    None = 0,
+    ServerError = 1,
+    ProtocolError = 2,
+    LoginEror = 3,
+    UntrustedDomain = 4,
+    NtlmNotConnected = 5,
+    SspiError = 6,
+    BadNtGroup = 7,
+    BadRequest = 8,
+    InternalError = 128,
+    BlobError = 129,
+    BadProtocol = 130
+};
 
 /** String header. String data resides at the end of the request */
-typedef struct _strhdr {
+typedef struct _strhdr
```

Comment:
You do not need to modify this code, but if you do modify it, then please drop `typedef` and `_strhdr`. Just use the regular C++ `class strhdr\n{\npublic: ...` template.

I recommend leaving these lines unchanged instead, to keep them consistent with other bad declarations in this file.

Reply:
undone - it was unintentional. I checked for other formatting change, found none